Source/Scene/Expression.js

```diff
@@ -127,31 +137,72 @@ define([
     };
 
     Node.prototype._evaluatePlus = function(feature) {
-        return this._left.evaluate(feature) + this._right.evaluate(feature);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        if (right instanceof Color && left instanceof Color) {
+            return Color.add(left, right, new Color());
+        }
+        return left + right;
     };
 
     Node.prototype._evaluateMinus = function(feature) {
-        return this._left.evaluate(feature) - this._right.evaluate(feature);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        if (right instanceof Color && left instanceof Color) {
+            return Color.subtract(left, right, new Color());
+        }
+        return left - right;
     };
 
     Node.prototype._evaluateTimes = function(feature) {
-        return this._left.evaluate(feature) * this._right.evaluate(feature);
+        var left = this._left.evaluate(feature);
+        var right = this._right.evaluate(feature);
+        if (right instanceof Color && left instanceof Color) {
+            return Color.multiply(left, right, new Color());
+        } else if (right instanceof Color && typeof(left) === 'number') {
+            return Color.multiplyByScalar(right, left, new Color());
+        } else if (left instanceof Color && typeof(right) === 'number') {
```

Comment:
When we do syntax errors, which we should start soon, we can check, for example, if `right` was a string here just once when walking the AST.